crates/cext/src/circuit.rs

```
@@ -908,11 +1034,11 @@ pub unsafe extern "C" fn qk_circuit_get_instruction(
     };
     let mut params = {
         let params = packed_inst.params_view();
-        let params_vec: Vec<f64> = params
+        let params_vec: Vec<*const Param> = params
             .iter()
             .map(|x| match x {
-                Param::Float(val) => *val,
-                _ => panic!("Invalid parameter on instruction"),
+                Param::Obj(_) => panic!("Invalid parameter on instruction"),
+                _ => x as *const Param,
```

Just as a minor point of style/safety, it might be nice to keep the precise matching for the happy paths and the wildcard for the failure, so we default safe in the face of `Param` changing, instead of defaulting to returning something that might cause `unsafe` problems in the C interface code?